generator/test_case_generator.go

```diff
@@ -6,17 +6,27 @@ package main
 import (
 	"encoding/json"
 	"fmt"
+	"github.com/mitchellh/mapstructure"
 	"io/ioutil"
 	"log"
 	"os"
 )
 
-const (
-	testDir = "test_dir"
-)
+type matrixLine struct {
+	TestDir             string `json:"testDir"`
+	Os                  string `json:"os"`
+	TestType            string `json:"testType"`
+	Arc                 string `json:"arc"`
+	InstanceType        string `json:"instanceType"`
+	Ami                 string `json:"ami"`
+	BinaryName          string `json:"binaryName"`
+	Username            string `json:"username"`
+	InstallAgentCommand string `json:"installAgentCommand"`
+	CaCertPath          string `json:"caCertPath"`
+}
```

Comment:
I'm assuming that by using the `mapstructure` package, we retain the ordering of the keys?